cmake/kokkos_arch.cmake

```diff
@@ -212,6 +212,9 @@ GLOBAL_SET(KOKKOS_AMDGPU_OPTIONS)
 IF(KOKKOS_ENABLE_HIP)
   SET(AMDGPU_ARCH_FLAG "--offload-arch")
   IF(NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
+    IF(KOKKOS_CXX_STANDARD STREQUAL 14 AND NOT CMAKE_CXX_STANDARD)
+      message(FATAL_ERROR "Set CMAKE_CXX_STANDARD to 14")
```

Comment:
The title line say "warning" but this is actually erroring out.  Did you mean to?

Reply:
I changed my mind after I created the PR and forgot to update the title.